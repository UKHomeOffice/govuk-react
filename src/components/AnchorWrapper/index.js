// https://govuk-elements.herokuapp.com/typography/

import React from "react";
import PropTypes from "prop-types";
import glamorous from "glamorous";
import { BLUE, PURPLE, YELLOW } from "govuk-colours";

const AnchorInner = glamorous.span({
  "> a": {
    color: BLUE,
    padding: "3px",
    margin: "-3px",
    outlineColor: "transparent",
    display: "inline-block",
    textDecoration: "underline",
    ":focus": {
      backgroundColor: YELLOW,
      outline: `3px solid ${YELLOW}`
    },
    ":visited": {
      color: PURPLE
    }
  }
});

const AnchorWrapper = ({ children }) => <AnchorInner>{children}</AnchorInner>;

AnchorWrapper.propTypes = {
  children: PropTypes.node.isRequired
};

export default AnchorWrapper;
