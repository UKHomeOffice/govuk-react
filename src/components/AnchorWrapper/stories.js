import React from "react";
import { storiesOf } from "@storybook/react";
import { BrowserRouter, Link } from "react-router-dom";
import AnchorWrapper from ".";

const exampleURL = "https://example.com";
const exampleChild = "example";

storiesOf("AnchorWrapper", module).add(
  "AnchorWrapper styling an anchor",
  () => (
    <AnchorWrapper>
      <a href={exampleURL}>{exampleChild}</a>
    </AnchorWrapper>
  )
);
storiesOf("AnchorWrapper", module).add(
  "AnchorWrapper styling a React Router Link",
  () => (
    <AnchorWrapper>
      <BrowserRouter>
        <Link to={exampleURL}>{exampleChild}</Link>
      </BrowserRouter>
    </AnchorWrapper>
  )
);
