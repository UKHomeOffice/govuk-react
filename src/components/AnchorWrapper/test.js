import React from "react";
import ReactDOM from "react-dom";
import { mount } from "enzyme";
import AnchorWrapper from "./";

const example = "example";
const wrapper = <AnchorWrapper hyperLink={`https://${example}.com`}>{example}</AnchorWrapper>;

describe(AnchorWrapper, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches wrapper snapshot", () => {
    expect(mount(wrapper)).toMatchSnapshot(`wrapper mount`);
  });
});
