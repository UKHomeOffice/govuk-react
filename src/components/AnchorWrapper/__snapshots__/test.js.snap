// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor matches wrapper snapshot: wrapper mount 1`] = `
.glamor-0,
[data-glamor-0] {
  color: #005ea5;
  padding: 3px;
  margin: -3px;
  outline-color: transparent;
  display: inline-block;
  text-decoration: underline;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  background-color: #ffbf47;
  outline: 3px solid #ffbf47;
}

.glamor-0:visited,
[data-glamor-0]:visited {
  color: #2e358b;
}

<Anchor
  hyperLink="https://example.com"
>
  <glamorous(a)
    href="https://example.com"
  >
    <a
      className="glamor-0"
      href="https://example.com"
    >
      example
    </a>
  </glamorous(a)>
</Anchor>
`;
